validator/sawtooth_validator/state/client_handlers.py

```diff
@@ -730,7 +730,7 @@ def _respond(self, request):
             entries,
             self._status.INVALID_PAGING)
 
-        if not entries:
+        if entries is None:
```

Comment:
This change doesn't make sense. If `entries` is the `None` object you will get a traceback on line 723 above b/c None object doesn't have a `.sort()` method. However the entries variable is assigned a list value just before that so the only place where it could become None is on line 728. 

From what I can see the method `.paginate_resources()` will not return None either so `entries` will continue to be a list, even if it is empty.

Reply:
@nathanwhit what is the call chain that gets us from `_find_address_values_in_chain()` to here ?

Reply:
You're right, `None` doesn't seem possible so that branch can be removed entirely.

Reply:
> @nathanwhit what is the call chain that gets us from `_find_address_values_in_chain()` to here ?

I don't think there is one, so you're right this isn't CRB-79. Thanks for pointing it out, I should've checked more carefully. I'll file a new ticket and update the PR title.

Reply:
The point of the PR is that we don't want the case of empty `entries` to be an error. When we're requesting all state entries by prefix (e.g. for iterating over all ask orders) and there are none, I'd expect to receive an empty list instead of an error.

Reply:
I suggest that, if there isn't strong evidence that this is a problem, leave it as it is and handle it explicitly in the tp.